src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -159,7 +159,7 @@ private void OnRunCodeAnalysisForSelectedProjectStatus(object sender, EventArgs
             if (visible)
             {
                 if (command.CommandID.ID == RunCodeAnalysisForSelectedProjectCommandId &&
-                    hierarchy.TryGetProject(out var project))
+                    hierarchy!.TryGetProject(out var project))
```

Comment:
So IsCapabilityMatch is an unannotated extension method in the Visual Studio shell, so it wouldn't flag no matter what. But TryGetSelectedProjectHierarchy is also annotated that it'll be non-null if it returns true, so since it's a && in line 155 and 156 that'll only run if non-null. But the compiler doesn't track smart enough to realize that TryGetSelectedProjectHierarchy returning true is required for visible to be true, which means this is safe.